liquidctl/driver/commander_core.py

```diff
@@ -0,0 +1,176 @@
+                status += [(f'Temperature Sensor {i}', 'Connected' if connected else 'Disconnected', '')]
+
+        return status
+
+    def get_status(self, **kwargs):
+        """Get all the fan speeds and temps"""
+        status = []
+
+        # INIT in case it hasn't been accesses in a while
+        self._send_command(_CMD_INIT)
+
+        for i, speed in self._get_speeds().items():
+            if i == 0:
+                status += [(f'Pump Speed', speed, 'rpm')]
+            else:
+                status += [(f'Fan Speed {i}', speed, 'rpm')]
+
+        for i, temp in self._get_temps().items():
+            if temp is None:
+                continue
+            if i == 0:
+                status += [(f'Water Temperature', temp, '°C')]
+            else:
+                status += [(f'Temperature {i}', temp, '°C')]
+
+        return status
+
+    def set_color(self, channel, mode, colors, **kwargs):
+        raise NotImplementedError
+
+    def set_speed_profile(self, channel, profile, **kwargs):
+        raise NotImplementedError
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
+        raise NotImplementedError
+
+    @classmethod
+    def probe(cls, handle, **kwargs):
+        """Ensure we get the right interface"""
+
+        if handle.hidinfo['interface_number'] == _INTERFACE_NUMBER:
+            return
+
+        yield from super().probe(handle, **kwargs)
+
+    def _get_speeds(self):
+        speeds = {}
+
+        self._send_command(_CMD_RESET)
+        self._send_command(_CMD_SET_MODE, [_MODE_GET_SPEEDS])
+        res = self._send_command(_CMD_GET)
+
+        num_speeds = res[5]
+        speeds_data = res[6:6 + num_speeds*2]
+        for i in range(0, num_speeds):
+            speeds[i] = u16le_from(speeds_data, offset=i*2)
+
+        return speeds
+
+    def _get_temps(self):
+        temps = {}
+
+        self._send_command(_CMD_RESET)
+        self._send_command(_CMD_SET_MODE, [_MODE_GET_TEMPS])
+        res = self._send_command(_CMD_GET)
+
+        num_temps = res[5]
+        temp_data = res[6:6 + num_temps*3]
+        for i in range(0, num_temps):
+            connected = temp_data[i*3] == 0x00
+            if connected:
+                temps[i] = u16le_from(temp_data, offset=i*3+1)/10
+            else:
+                temps[i] = None
+
+        return temps
+
+    def _send_command(self, command, data=()):
+        # self.device.write expects buf[0] to be the report number or 0 if not used
+        buf = bytearray(_REPORT_LENGTH + 1)
+
+        # buf[1] when going out is always 08
+        buf[1] = 0x08
+
+        # Indexes for the buffer
+        cmd_start = 2
+        data_start = cmd_start + len(command)
+        data_end = data_start + len(data)
+
+        # Fill in the buffer
+        buf[cmd_start: data_start] = command
+        buf[data_start:data_end] = data
+
+        self.device.clear_enqueued_reports()
+        self.device.write(buf)
+
+        buf = bytes(self.device.read(_RESPONSE_LENGTH))
```

Comment:
Is there something you can assert in the reply to check that the command was successful?

Reply:
Depends on what you mean by successful.
Let's say command `0x01` failed, it would recive `00:08:01` and then the rest would be data from the last responce we've recived as it did not clear it.
Most I think we could do is `assert buf[2] == command[0]` to ensure that the resonce we recived is for that command.

Reply:
I see, so there really is no error checking in the protocol.

> Most I think we could do is assert buf[2] == command[0] [...]

Yeah, I think that could be (marginally) useful, please add it.

Reply:
was this supposed to have been added?

Reply:
be7a9734108cf7d3021782262ceb4686706f8e4a